### PR TITLE
OSDOCS#7408: Adding autogenerated cli tools to MicroShift

### PR DESCRIPTION
**Pull request by kelbrown20**
**Version(s):** 4.14+
**Issue:** [OSDOCS-7408](https://issues.redhat.com//browse/OSDOCS-7408)

**Link to docs preview:**

[OpenShift CLI command reference](https://65702--docspreview.netlify.app/microshift/latest/microshift_cli_ref/microshift-oc-cli-commands-list)

**QE review:**
- [ ] QE has approved this change.

### CODE REVIEW

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 1]
```diff
@@ -18,101 +16,181 @@ Mirror an operator-registry catalog
...
+  oc adm ocp-certificates monitor-certificates
+----
+
 
-  # See what logs are available in masters in /var/logs
-  oc adm node-logs --role master --path=/
 
-  # Display cron log file from all masters
-  oc adm node-logs --role master --path=cron
+== oc adm ocp-certificates regenerate-leaf
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:100]: Remove: I was not able to get this one working

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 2]
```diff
@@ -18,101 +16,181 @@ Mirror an operator-registry catalog
...
+  # Regenerate all known leaf certificates on the cluster.
+  oc adm certificates regenerate-leaf -A --all
+  
+  # Regenerate a leaf certificate contained in a particular secret.
+  oc adm ocp-certificates regenerate-leaf -n openshift-config-managed kube-controller-manager-client-cert
+----
+
+
+
+== oc adm ocp-certificates regenerate-machine-config-server-serving-cert
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:115]: Remove: we don't have machine configs

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 3]
```diff
@@ -18,101 +16,181 @@ Mirror an operator-registry catalog
...
+  # Regenerate the MCO certs without modifying user-data secrets
+  oc adm certificates regenerate-machine-config-server-serving-cert --update-ignition=false
+  
+  # Update the user-data secrets to use new MCS certs
+  oc adm certificates update-ignition-ca-bundle-for-machine-config-server
+----
+
+
+
+== oc adm ocp-certificates regenerate-top-level
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:130]: Remove: couldn't get this succeed

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 4]
```diff
@@ -18,101 +16,181 @@ Mirror an operator-registry catalog
...
+  # Regenerate all known signing certificates on the cluster.
+  oc adm certificates regenerate-signers -A --all
+  
+  # Regenerate the signing certificate contained in a particular secret.
+  oc adm certificates regenerate-signers -n openshift-kube-apiserver-operator loadbalancer-serving-signer
+----
+
+
+
+== oc adm ocp-certificates remove-old-trust
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:145]: Remove: This doesn't error, but I'm not seeing any change in configmaps, safe to remove from our docs for now

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 5]
```diff
@@ -18,101 +16,181 @@ Mirror an operator-registry catalog
...
+  # Remove a trust bundled contained in a particular config map
+  oc adm ocp-certificates remove-old-trust -n openshift-config-managed configmaps/kube-apiserver-aggregator-client-ca
+  
+  #  Remove only CA certificates created before a certain date from all trust bundles
+  oc adm ocp-certificates remove-old-trust configmaps -A --all --created-before 2023-06-05T14:44:06Z
+----
+
+
+
+== oc adm ocp-certificates update-ignition-ca-bundle-for-machine-config-server
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:163]: Remove: we don't use ignition and machine config

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 6]
```diff
@@ -18,101 +16,181 @@ Mirror an operator-registry catalog
...
+  # Regenerate the MCO certs without modifying user-data secrets
+  oc adm certificates regenerate-machine-config-server-serving-cert --update-ignition=false
+  
+  # Update the user-data secrets to use new MCS certs
+  oc adm certificates update-ignition-ca-bundle-for-machine-config-server
+----
+
+
+
+== oc adm reboot-machine-config-pool
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:178]: Remove: we don't use machine config pools

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 7]
```diff
@@ -168,25 +246,45 @@ Mirror a release to a different image registry location
...
+  
   # Mirror the 4.11.0 release to repository registry.example.com and apply signatures to connected cluster
   oc adm release mirror --from=quay.io/openshift-release-dev/ocp-release:4.11.0-x86_64 \
   --to=registry.example.com/your/repository --apply-release-image-signature
 ----
 
 
-//== oc adm release new
+
+== oc adm restart-kubelet
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:268]: Remove: does not work and we have kubelet is embedded into microshift

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 8]
```diff
@@ -197,25 +295,21 @@ Update the taints on one or more nodes
...
   oc adm taint nodes foo bar:NoSchedule
 ----
 
 
-//== oc adm top images
-//== oc adm top imagestreams
-//== oc adm top node
-
 
 == oc adm top pod
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:314]: Remove: metrics api is not turned on

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 9]
```diff
@@ -225,17 +319,48 @@ Display resource (CPU/memory) usage of pods
...
   # Show metrics for the pods defined by label name=myLabel
   oc adm top pod -l name=myLabel
 ----
 
-//== oc adm uncordon
-//== oc adm upgrade
-//== oc adm verify-image-signature
+
+
+== oc adm wait-for-node-reboot
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:335]: Remove: no machine config pools

File: modules/microshift-oc-adm-by-example-content.adoc [hunk 10]
```diff
@@ -225,17 +319,48 @@ Display resource (CPU/memory) usage of pods
...
+  # Wait for masters to complete a requested reboot from 'oc adm reboot-machine-config-pool mcp/master'
+  oc adm wait-for-node-reboot nodes -l node-role.kubernetes.io/master
+  
+  # Wait for masters to complete a specific reboot
+  oc adm wait-for-node-reboot nodes -l node-role.kubernetes.io/master --reboot-number=4
+----
+
+
+
+== oc adm wait-for-stable-cluster
```
Comment (eggfoobar) [inline modules/microshift-oc-adm-by-example-content.adoc:353]: Remove: needs cluster operators

**Review by eggfoobar — COMMENTED** [review]
Identified the ones we can safely remove, either they do not work or use a config that the command relies on.

**Comment by kelbrown20** [comment]
PR no longer needed

**Pull request closed without merging**